http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnector.java

```diff
@@ -0,0 +1,77 @@
+package org.triplea.http.client.web.socket;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.java_websocket.client.WebSocketClient;
+
+/**
+ * Class to handle the task of creating a websocket connection asynchronously and then provides a a
+ * blocking method to allow us to wait for the connection to be established. Connecting to a local
+ * server, a websocket connection can take several hundred milliseconds.
+ */
+class WebSocketConnector {
+  private static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = 5000;
+
+  private final WebSocketClient webSocketClient;
+  private final int connectTimeoutMillis;
+
+  /**
+   * We use this to create a single blocking task that is our connection attempt. Eventually to
+   * block for completion, we'll await completion of tasks on this queue.
+   */
+  private ExecutorService executorService;
+
+  /**
+   * This is the thread that is attempting to connect to server. Returns true if connection is
+   * successful otherwise false.
+   */
+  private Future<Boolean> connectionThread;
+
+  WebSocketConnector(final WebSocketClient webSocketClient) {
+    this(webSocketClient, DEFAULT_CONNECT_TIMEOUT_MILLIS);
+  }
+
+  @VisibleForTesting
+  WebSocketConnector(final WebSocketClient webSocketClient, final int connectTimeoutMillis) {
+    this.webSocketClient = webSocketClient;
+    this.connectTimeoutMillis = connectTimeoutMillis;
+  }
+
+  void initiateConnection() {
+    executorService = Executors.newSingleThreadExecutor();
+    connectionThread = executorService.submit((Callable<Boolean>) webSocketClient::connectBlocking);
+    executorService.shutdownNow();
+  }
+
+  void waitUntilConnectionIsOpen() {
+    try {
+      executorService.awaitTermination(connectTimeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
Ok, actually didn't think of it this way.
Now you're really using the blocking method + executor as a fancy way to create some sort of latch to block other threads.
Definitely something I could live with, but I mean a `CountDownLatch` +a custom on open implementation of the connection is opened asynchronously would do the trick as well with less complexity.

Reply:
How do you incorporate the timeout in that case? 

Of concern, let's say we have a timeout and we queue messages to send in the same threadpool. If the time to flush all messages takes longer than the timeout, then we'll have an incorrect error not because the connection could not be opened, but because instead we did not send messages in time.

To some extent the original code, though a bit manual, is starting to look not too bad given the cleverness and interplay to avoid having a manual, blocking, timeout. I'm starting to ponder if we hit diminishing returns.